### PR TITLE
Feat/event creation offline

### PR DESCRIPTION
**Pull request by Zafouche**
For full offline mode feature overview, see [this issue](https://github.com/SwEnt-Group13/Unio/issues/228).

For this specific feature, see the full description and the choice of solution behind it in [this issue](https://github.com/SwEnt-Group13/Unio/issues/248).

## Description
This PR implements a simple handling of the add event button in offline mode. When we click on it, it prevents the user from navigating to the event creation screen, which means the user isn't allowed to create one, just like a user on any social media wouldn't be allowed to create a post while offline.

## Motivation and context
This fix goes in the spirit of adapting our app's features for offline mode. This was one of the features that needed additional measures to respect it.

### AssociationProfile: 
- Added isOnline boolean that checks if the user is online or not.
- Added logic to the `Add event` button to display a toast that says the user is offline if it is the case.

## How has this been tested?

In AssociationProfileTest, I added two tests: 
- testAddEventButtonOnline: checks that the button click takes the user to the event creation screen.
- testAddEventButtonOffline: checks that the button click does *not* take the user to the event creation screen.

### CODE REVIEW

File: app/src/androidTest/java/com/android/unio/components/association/AssociationProfileTest.kt [hunk 1]
```diff
@@ -338,11 +341,9 @@ class AssociationProfileTest : TearDown() {
           navigationAction, userViewModel, eventViewModel, associationViewModel) {}
     }
 
-    `when`(navigationAction.navController.popBackStack()).thenReturn(true)
-
     composeTestRule.onNodeWithTag(AssociationProfileTestTags.GO_BACK_BUTTON).performClick()
 
-    verify(navigationAction.navController).popBackStack()
+    verify { navigationAction.goBack() }
```
Comment (AlouchLaBouche) [inline app/src/androidTest/java/com/android/unio/components/association/AssociationProfileTest.kt:346]: Nice fix :+1:

File: app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt [hunk 2]
```diff
@@ -496,7 +497,15 @@ private fun AssociationEvents(
   }
   if (isAdmin) {
     Button(
-        onClick = { navigationAction.navigateTo(Screen.EVENT_CREATION) },
+        onClick = {
```
Comment (AlouchLaBouche) [inline app/src/main/java/com/android/unio/ui/association/AssociationProfile.kt:500]: Nice modification to ensure that the user is connected for before adding events !

**Review by AlouchLaBouche — APPROVED** [review]
Looks good to me! Nice changes that address the offline mode! Great work :rocket:

**Comment by Zafouche** [comment]
> Looks good to me! Nice changes that address the offline mode! Great work 🚀

Thanks for being quick to review!

**Pull request merged**